querydsl-core/src/main/java/com/querydsl/core/types/ExpressionUtils.java

```diff
@@ -533,14 +533,14 @@ public static Predicate or(Predicate left, Predicate right) {
     }
 
     /**
-     * Create a new root variable based on the given path
+     * Create a new root variable based on the given path and suffix
      *
      * @param path
+     * @param suffix
      * @return
      */
-    public static String createRootVariable(Path<?> path) {
+    public static String createRootVariable(Path<?> path, int suffix) {
         String variable = path.accept(ToStringVisitor.DEFAULT, TEMPLATES).replace('.', '_');
```

Comment:
The templates used are already UnderscoreTemplates, so is it necessary to replace '.' with '_'?